### PR TITLE
958 integration test reorganization

### PR DESCRIPTION
**Pull request by BradleyBoutcher**
#### What does this PR do (include background context, if relevant)?
- Restructures test directory to match source code structure and make test location simpler
- Improves Jenkins script to programmatically create integration stages for new directories, running in parallel
- Improves run_integration script to lookup test directory, regardless of depth
(Note: I think these two changes could be improved. It may be more efficient for Jenkins to pass the script a full directory, rather than shortening it first) 

#### What ticket does this PR close?
Connected to #958 

#### What is the status of the manual tests?
Have you run the following manual tests to verify existing functionality continues to function as expected?
- [X] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/manual/keychain_provider)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

If this feature does not have any/sufficent automated tests, have you created/updated a folder in `test/manual` that includes:
- [ ] An updated README with instructions on how to manually test this feature
(Note: We may need to update the test related style guide)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -18,13 +18,42 @@ pipeline {
...
+            script: 
+            '''
+            find $(find ./test -name test) -name 'start' -exec dirname {} \\; | xargs -n1 basename
+            '''
+          ).trim().split()
+
+          def integrationSteps = [:]
+        
+          directories.each { name -> 
+            if (name == "keychain") return
```
Comment (sgnn7) [inline Jenkinsfile]: Needs a comment on why

File: Jenkinsfile [hunk 2]
```diff
@@ -18,13 +18,42 @@ pipeline {
...
     }
 
+    stage('Integration Tests') {
+      steps { 
+        script {
+          def directories = sh (
+            returnStdout: true,
+            script: 
+            '''
+            find $(find ./test -name test) -name 'start' -exec dirname {} \\; | xargs -n1 basename
```
Comment (sgnn7) [inline Jenkinsfile]: Should probably have a comment about what it's searching for

File: test/connector/http/generic/docker-compose.yml [hunk 3]
```diff
@@ -32,4 +32,4 @@ services:
   dev:
     image: secretless-dev
     volumes:
-      - ../..:/secretless
+      - ../../../..:/secretless
```
Comment (sgnn7) [inline test/connector/http/generic/docker-compose.yml]: Per slack convo we should at some point figure this out - it's needlessly complicated of a path

File: test/connector/tcp/mysql/docker-compose.yml [hunk 4]
```diff
@@ -56,7 +56,7 @@ services:
       dockerfile: Dockerfile.dev
     command: sleep 999d
     environment:
-      TEST_ROOT: /secretless/test/mysql_handler
+      TEST_ROOT: /secretless/test/connector/tcp/mysql
```
Comment (sgnn7) [inline test/connector/tcp/mysql/docker-compose.yml]: It's not a blocker but keep in mind that integration tests usually cover much more functionality than just the module they are testing so this mirrored nesting under `test` might be overkill

**Comment by BradleyBoutcher** [comment]
> @BradleyBoutcher Just a few nits. There are also some wrapping issues w/ your git commit descriptions (line-too-long) so make sure those are <100 chars but it's not a blocker.

I went through and edited my commits to make sure they wrap - good catch

File: bin/run_integration [hunk 5]
```diff
@@ -49,7 +49,7 @@ export COMPOSE_PROJECT_NAME
 # Validate the required scripts exists in the TEST_SUBDIR
 
 project_dir=$PWD
-test_dir="${project_dir}/test/${non_flag_args[0]}"
+test_dir="$(find "${project_dir}"/test -type d -name "${non_flag_args[0]}")"
```
Comment (jonahx) [inline bin/run_integration]: Small thing and I know this is existing code, but as a general rule when editing a file for a new feature or bug, we try to correct previous mistakes we've made, especially when it's quick to do so:

In this case, the `non_flag_args` is impressively uninformative.  Let's rename to `test_subdirectory` or similar.

File: test/connector/http/basicauth/docker-compose.yml [hunk 6]
```diff
@@ -15,14 +15,14 @@ services:
 
   secretless:
     build:
-      context: ../..
+      context: ../../../..
```
Comment (jonahx) [inline test/connector/http/basicauth/docker-compose.yml]: Let's make sure to add a ZH issue for improving this per our earlier discussion.  If you've already done so, just ignore this comment.

**Pull request merged**